### PR TITLE
Fixes #23635 - Centralize use of facter

### PR DESCRIPTION
**Pull request by ekohl**
We have a bundler group for facter. That implies it's optional but
before this change it wasn't With this change it's only required if no
domain and fqdn are set. This is still the default so in most cases it's
no difference, but at least we have centralized all use of facter to one
place.

### CODE REVIEW

**Review by sean797 — CHANGES_REQUESTED** [review]
Mind also removing https://github.com/theforeman/foreman/blob/d73ddd4fa6068f846dae2486d2e5611b201ce49d/app/models/setting/email.rb#L1

**Comment by ekohl** [comment]
You're right, I missed a few.

**Comment by ekohl** [comment]
Updated to remove more `require 'facter'` statements.

**Review by sean797 — APPROVED** [review]
ACK pending tests. Thanks @ekohl  :+1:

**Comment by ekohl** [comment]
Looks like Katello has some assumptions that Facter is present. Will send a PR.

**Comment by ekohl** [comment]
https://github.com/Katello/katello/pull/7384

**Review by timogoebel — APPROVED** [review]
This can be merged once the Katello PR is merged.

**Comment by timogoebel** [comment]
[test katello]

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

**Pull request merged**
